client/js/router.js

```diff
@@ -32,12 +32,58 @@ const router = new VueRouter({
 				next();
 			},
 		},
+		{
+			name: "Connect",
+			path: "/connect",
+			component: Connect,
+			props: (route) => ({queryParams: route.query}),
+		},
+		{
+			name: "Settings",
+			path: "/settings",
+			component: Settings,
+		},
+		{
+			name: "Help",
+			path: "/help",
+			component: Help,
+		},
+		{
+			name: "Changelog",
+			path: "/changelog",
+			component: Changelog,
+		},
+		{
+			name: "NetworkEdit",
+			path: "/edit-network/:uuid",
+			component: NetworkEdit,
+		},
+		{
+			name: "RoutedChat",
+			path: "/chan-:id",
+			component: RoutedChat,
+		},
 	],
 });
 
+router.beforeEach((to, from, next) => {
+	// If user is not yet signed in, wait for appLoaded state to change
+	// unless they are trying to open SignIn (which can be triggered in auth.js)
+	if (!store.state.appLoaded && to.name !== "SignIn") {
+		store.watch(
+			(state) => state.appLoaded,
+			() => next()
```

Comment:
The only "issue" here is that it is possible to navigate to signin before this `next` happens.

If you reload the page with `/settings` and your auth fails, it will technically happen later after sign in here.

I haven't noticed it actually breaking anything though.